### ISSUE TITLE
Parallel solver with KLU

### ISSUE THREAD

**Issue by Amuri44** [JuliaSparse/KLU.jl#7]
Is there any parallel solver for KLU sofar, such as that for MKLPardisoSolver?

**Comment by Wimmerer** [comment]
For the solve step? You mean like parallelizing over a matrix solve (one thread per vector for instance)?

**Comment by Amuri44** [comment]
Yes, exactly. Which would be wonderful. Cause I did not know how to start and what are the steps.

**Comment by mzy2240** [comment]
Definitely nice to have! This new [one](https://github.com/chenxm1986/cktso) may be similar to Basker as well.

**Comment by Amuri44** [comment]
> Frankly speaking, factorization in KLU is already fast enough, however everytime when I try to use the same factorized matrix across multiple threads it already slow down the whole process.

Excuse me, if we need to make the solution of the linear equation `Ax=b` on two threads, so what is the steps you do with the factorized matrix? In other words, after the factor step (on serial) what do you do? cause I did not know how to solve this `x=(LU+F)\b` comparing with `x=U\L\b` in LU factorization.

**Comment by mzy2240** [comment]
In my use case, b is multi-vector or 2D matrix whatever you name it, so apparently the solve process could be paralleled.

**Comment by Amuri44** [comment]
> In my use case, b is multi-vector or 2D matrix whatever you name it, so apparently the solve process could be paralleled.

I see. So, each thread will work on one vector or `b` but with the same `A`, right?

**Comment by mzy2240** [comment]
Yes

**Comment by Amuri44** [comment]
@Wimmerer Your help here is very appreciated.
Given a linear equation `Ax=b:
-  The forward and backward substitution in LU factorization is straightforward, `x=U\L\b`
-  However, what is the forward and backward substitution in KLU factorization, cause this way  `x=(LU+F)\b` seems to be very slow.
